macros/src/main/scala-3/com/softwaremill/macwire/MacwireMacros.scala

```diff
@@ -2,16 +2,40 @@ package com.softwaremill.macwire
 
 import com.softwaremill.macwire.internals.*
 import scala.quoted.*
-import java.sql.Types
 
 object MacwireMacros {
   private val log = new Logger()
 
   def wireImpl[T: Type](using q: Quotes): Expr[T] = {
     import q.reflect.*
 
-    val constructorCrimper = new ConstructorCrimper[q.type, T](log)
-    val companionCrimper = new CompanionCrimper[q.type, T](log)
+    val dependencyResolver = DependencyResolver.throwErrorOnResolutionFailure[q.type, T](log)
+
+    wire[T](using q)(dependencyResolver)
+  }
+
+  //TODO build failure path 
+  def wireRecImpl[T: Type](using q: Quotes): Expr[T] = {
+    import q.reflect.*
+
+    // FIXME for some reason `TypeRepr.of[String].typeSymbol.owner` and `defn.JavaLangPackage` have different hash codes 
+    def isWireable(tpe: TypeRepr): Boolean = tpe.classSymbol.map(_.owner.fullName != defn.JavaLangPackage.fullName).getOrElse(false)
+    
+    val dependencyResolver = new DependencyResolver[q.type, T](using q)(log, tpe => 
```

Comment:
maybe you could create a method like `throwErrorOnResolutionFailure` on `DependencyResolver` as well, but creating the recursive one?

Reply:
I like that idea, but in the recursive case of the `resolutionFallback` we refer to the `wireRec` method directly, which means we need to define `recusiveResoultion` on `DependencyResolver` like:
```
  def recursiveResolution[Q <: Quotes, T: Type](debug: Logger)(using q: Q)(recCall: q.reflect.TypeRepr => q.reflect.Term) = new DependencyResolver[q.type, T](using q)(debug, tpe => 
      if !isWireable(tpe) then q.reflect.report.throwError(s"Cannot find a value of type: [${showTypeName(tpe)}]")
      else recCall(tpe)
    )
```

so I would rather stick with the current solution.